docs/howto/partitions.rst

```diff
@@ -0,0 +1,167 @@
+   class about our partitions, but applications do not usually directly
+   instantiate the LifecycleManager.
+
+   Instead, override your :class:`Application
+   <craft_application.Application>`'s ``_setup_partitions`` method, and return
+   a list of the partitions, which will eventually be passed to the
+   :class:`LifecycleManager <craft_parts.LifecycleManager>`:
+
+   .. code-block:: python
+
+     class SnackcraftApplication(Application):
+
+       ...
+
+       @override
+       def _setup_partitions(self, yaml_data: dict[str, Any]) -> list[str] | None:
+           return ["default", "kernel", "component/bar-baz"]
+
+Using the partitions
+====================
+
+Partitions cannot be used until `after you have configured your application
+<#app-changes>`_.
+
+In configuration yaml
+---------------------
+
+Defined partitions may be referenced in the ``organize``, ``stage``, and
+``prime`` sections of your configuration yaml files:
+
+.. code-block:: yaml
+
+  organize:
+    <source-path>: (<partition>)/<path>
+  stage:
+    - (<partition>)/<path>
+  prime:
+    - (<partition>)/<path>
+
+Paths in the configuration yaml not beginning with a partition label will
+implicitly use the default partition.
+
+The source path of an ``organize`` entry can only be from the default
+partition.  For example:
+
+.. code-block:: yaml
+
+  organize:
+    (kernel)/usr/local/bin/hello: bin/hello
+
+.. code-block:: text
+
+  Cannot organize files from 'kernel' partition.
+  Files can only be organized from the 'default' partition
+
+When the ``stage`` and ``prime`` keywords are not provided for a part,
+craft-parts' default behavior is to stage and prime all files for the part in
+all partitions.
+
+(If a stage or prime filter *is* applied to a partition, the default behavior
+will not be affected for the other partitions.)
+
+See also
+^^^^^^^^
+
+ * Craft parts: part properties: `organize`_
+ * Craft parts: filesets: `specifying paths`_
+
+In environment variables
+------------------------
+
+You might use these variables in a lifecycle override section of a
+configuration yaml.  For instance:
+
+.. code-block:: yaml
+
+  override-prime: |
+    cp -R $CRAFT_KERNEL_STAGE/vmlinux $CRAFT_KERNEL_PRIME/
+    chmod -R 444 $CRAFT_KERNEL_PRIME/*
+    cp -R $CRAFT_STAGE/lib/modules/6.x/* $CRAFT_PRIME
+    chmod -R 600 $CRAFT_PRIME/*
+
+See also
+^^^^^^^^
+
+ * Craft parts: parts and steps: `environment variables`_
+ * Craft parts: part properties: `override-prime`_
+
+In code
+-------
+
+Application code that can access ``Part`` or ``ProjectDirs`` objects may get
+partition information from them:
+
+.. code-block:: python-console
+
+  >>> Part(name="my-part").part_install_dirs["kernel"]
+  Path("partitions/kernel/parts/my-part/install")
+
+  >>> ProjectDirs.get_stage_dir(partition="kernel")
+  Path("/root/partitions/kernel/stage")
+
+  >>> ProjectDirs.get_prime_dir(partition="component/bar-baz")
+  Path("/root/partitions/component/bar-baz/prime")
```

Comment:
These will not work, the methods are not classmethods.

Reply:
I think the recommended way to do this is via the Lifecycle service's `project_info` [property](https://github.com/canonical/craft-application/blob/main/craft_application/services/lifecycle.py#L210)

Reply:
Thanks @tigarmo, that was helpful.

Reply:
Actually, @tigarmo, a related question - is the example before this one (`Part(name="my-part").part_install_dirs["kernel"]`) useless?  It doesn't look like there's any way for applications to get access to `Part` objects.

Reply:
Good point; typically the application doesn't/shouldn't really care about each part's individual directories, so maybe that bit should be removed

Reply:
https://github.com/canonical/craft-application/pull/371/commits/c915fdfcfec9e201cb8f8b7302fe3b08b86ea834